### DISCUSSION TITLE
Do not store secrets in plain text

### DISCUSSION THREAD

**Discussion by RaHehl** [hoppscotch/hoppscotch#3404] [Ideas]
As far as I am informed, today all data is stored in plain text, both in local storage and, actually, even more serious is storing it on the server, be it on premise or in the cloud.

In discussion #3020 there was no contrary statement regarding "what is their security"

In my case we have more than 300,000 employees, so internal hosting is far from enough to protect the credentials used and stored.

Both hoppscotch.io and self-hosted are, in principle, a gigantic honeypot for evil activities without data encryption at the user/team level.

To make a suggestion: environments, but better also the collections, should be encrypted user/team wise.

With an additional password per team/user which is not known to the backend and only for client-side encryption and decryption.

**Comment by GetPsyched** [comment]
Agreed; secrets not being encrypted at rest is very dangerous.

**Comment by marcellusherlus** [comment]
Any updates here? This hinders us to use hoppscotch in our company right now :(

**Reply by mirarifhasan** [reply]
We upgraded the flow. We're now storing secret data after encryption.
